godot-ffi/src/godot_ffi.rs

```diff
@@ -94,6 +94,59 @@ pub unsafe trait GodotFfi {
     unsafe fn move_return_ptr(self, dst: sys::GDExtensionTypePtr, call_type: PtrcallType);
 }
 
+/// Marks a type as having a nullable counterpart in Godot.
+///
+/// # Safety
+///
+/// The type has to have a pointer sized counter part in Godot which needs to be nullable.
+/// So far only types that inherit from Object are nullable.
+pub unsafe trait GodotNullablePtrSized: GodotFfi {}
+
+unsafe impl<T> GodotFfi for Option<T>
+where
+    T: GodotNullablePtrSized,
+{
+    fn sys(&self) -> sys::GDExtensionTypePtr {
+        match self {
+            Some(value) => value.sys(),
+            None => {
+                let nil_ptr = ptr::null_mut();
+                let new_nil = interface_fn!(variant_new_nil);
+
+                unsafe {
+                    new_nil(nil_ptr);
+                }
+
+                nil_ptr as sys::GDExtensionTypePtr
```

Comment:
does this not work if you just return a null pointer directly? (i.e just return `ptr::null_mut()`)

Reply:
It does.